src/Sound/SoundPack.cpp

```diff
@@ -0,0 +1,106 @@
+/*****************************************************************************
+ * Copyright (c) 2018-2020 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "SoundPack.h"
+
+#include <Common/FileSystem.h>
+#include <Game.h>
+#include <spdlog/spdlog.h>
+
+namespace openblack::audio
+{
+
+void SoundPack::LoadFromFile(const std::string& filename)
+{
+	spdlog::debug("Loading Sound Pack from file: {}", filename);
+
+	try
+	{
+		_sadFile.Open(Game::instance()->GetFileSystem().FindPath(filename).u8string());
+	}
+	catch (std::runtime_error& err)
+	{
+		spdlog::error("Failed to open {}: {}", filename, err.what());
+		return;
+	}
+
+	if (!_sadFile.GetName().empty())
+		_name = _sadFile.GetName();
+	else
+		_name = (filename);
+
+	// TODO: Discover sector flag
+	bool isSector;
+	auto combinedData = std::map<std::string, Sound>();
```

Comment:
SadFile is duplicating what PackFile does.
You should be using PackFile to extract the segments and SadFile should support only reading these segments.
See how src/3D/AnimationPack.cpp does it

Reply:
SadFile has been removed. PackFile now does everything